ArmPkg/Drivers/SmmuDxe/SmmuV3.h

```diff
@@ -0,0 +1,227 @@
+/** @file SmmuV3.h
+
+    This file is the SmmuV3 header file for SMMU driver compliant with the Smmu spec:
+
+    <https://developer.arm.com/documentation/ihi0070/latest/>
+
+    Copyright (c) Microsoft Corporation.
+    SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef SMMUV3_H_
+#define SMMUV3_H_
+
+#include "SmmuV3Registers.h"
+
+#define ROUND_UP(_Value_, _Alignment_) \
+    (((_Value_) + (_Alignment_) - 1) & ~((_Alignment_) - 1))
+
+//
+// Macros to align values up or down. Alignment is required to be power of 2.
+//
+
+#define ALIGN_DOWN_BY(length, alignment) \
+    ((UINT64)(length) & ~((UINT64)(alignment) - 1))
+
+#define ALIGN_UP_BY(length, alignment) \

```

Comment:
Try to use some of the existing align macros where possible https://github.com/tianocore/edk2/blob/master/MdePkg/Include/Base.h#L948.

Reply:
done